SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialNetDriverLoadBalancingHandler.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialNetDriverLoadBalancingHandler.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+
+FSpatialNetDriverLoadBalancingContext::FSpatialNetDriverLoadBalancingContext(USpatialNetDriver* InNetDriver, TArray<FNetworkObjectInfo*>& InOutNetworkObjects)
+	: NetDriver(InNetDriver)
+	, NetworkObjects(InOutNetworkObjects)
+{
+
+}
+
+void FSpatialNetDriverLoadBalancingContext::UpdateWithAdditionalActors()
+{
+	// Processing actors that we pull from actor hierarchies, in order to migrate them together.
+	// This loop is extracted from UNetDriver::ServerReplicateActors_BuildNetworkObjects
+	for (auto ActorInfoIter = AdditionalActorsToReplicate.CreateIterator(); ActorInfoIter; ++ActorInfoIter)
+	{
+		FNetworkObjectInfo* ActorInfo = *ActorInfoIter;
+		AActor* Actor = ActorInfo->Actor;
+
+		if (Actor->IsPendingKillPending())
+		{
+			continue;
+		}
+
+		// Verify the actor is actually initialized (it might have been intentionally spawn deferred until a later frame)
+		if (!Actor->IsActorInitialized())
+		{
+			continue;
+		}
+
+		// Don't send actors that may still be streaming in or out
+		ULevel* Level = Actor->GetLevel();
+		if (Level->HasVisibilityChangeRequestPending() || Level->bIsAssociatingLevel)
+		{
+			continue;
+		}
+
+		if (Actor->NetDormancy == DORM_Initial && Actor->IsNetStartupActor())
+		{
+			continue;
+		}
```

Comment:
I realise these these are copied from `BuildNetworkObjects` but should they be here vs. earlier when we're iterating through Actor hierarchies in the first place? If any of these are true, we should likely avoid iterating all of the children in `GetLatestAuthorityChangeFromHierarchy`?

Reply:
If any of this is true, it would be unlikely for these actors to have an actor channel, hence GetLatestAuthorityChangeFromHierarchy would likely not process them.
But it's true we could check that as well when browsing the hierarchy, and add a filter method for CollectActorsToMigrate.

Reply:
Moved these checks to IsActorReadyForMigration, which is called on hierarchy actors.